### DISCUSSION TITLE
How do I display the probability of that each type in the bounding box?

### DISCUSSION THREAD

**Discussion by doubletaro** [ultralytics/yolov5#7835] [Q&A] [answered]
For example, is it possible to display each probability in one bounding box as shown below?

cat: 30%
dog: 15%
squirrel: 5%

**Comment by glenn-jocher** [comment, marked as answer]
@doubletaro no. You can access these inside the NMS function:
https://github.com/ultralytics/yolov5/blob/d29df68299a216abb3254752bbf85e28fc92e737/detect.py#L120-L129

**Reply by doubletaro** [reply]
@glenn-jocher 
I have confirmed the NMS function!

Does "x[:, 5:]" on line 810 represent the probability of each category item?


https://github.com/ultralytics/yolov5/blob/614ef110b395eec842feac55f38805161f6ddb2e/utils/general.py#L805-L811

![01](https://user-images.githubusercontent.com/45992024/168732455-48c33bbf-6763-416f-af52-9e7449f57d72.png)

Is the above 24.4% for cats and 1.5% for dogs?

**Reply by doubletaro** [reply]
tensor([[0.24416, 0.01506],
        [0.25742, 0.02184],
        [0.38373, 0.02306],
        [0.33808, 0.03491],
        [0.39238, 0.01922],
        [0.48515, 0.03337],
        [0.51681, 0.02425],
        [**0.56923**, **0.04750**]], device='cuda:0')

To be more precise, the highest number is the probability of a bounding box, isn't it?
In this case, is it as follows?

cat:56.9%
dog:4.7%

**Reply by glenn-jocher** [reply]
Yes x[:, 5:] are the class probabilities

**Reply by doubletaro** [reply]
@glenn-jocher 
Thank you for your help!

**Comment by doubletaro** [comment]
oh.
Tomorrow I will check the function of NMS.
If I don't know something, please tell me again!
thank you!
